apps/address-book/app/store/entry.js

```diff
@@ -29,20 +30,43 @@ export const onEntryRemoved = async ({ entries }, { addr }) => {
   return entries
 }
 
+export const onEntryUpdated = async ({ entries = [] }, { addr }) => {
+  const index = entries.indexOf(entry => entry.addr === addr)
+
+  if (index) {
+    const data = await loadEntryData(addr) // async load data from contract
+    if (data) { // just perform transform if data was found (entry was not subsequently removed)
+      const entry = { addr, data }
+      entries[index] = entry
+    }
+  } else {
+    onEntryAdded({ entries }, { addr })
+  }
+
+  // return the (un)modified entries array
+  return entries
+}
+
 /// /////////////////////////////////////
 /*    AddressBook helper functions    */
 /// /////////////////////////////////////
 
 const loadEntryData = (addr) => {
-  return new Promise(resolve => {
-    app.call('getEntry', addr).subscribe(entry => {
-      // don't resolve when entry not found
-      entry &&
-        resolve({
-          entryAddress: entry[0],
-          name: entry[1],
-          entryType: entry[2],
+  return new Promise((resolve, reject) => {
+    app.call('getEntry', addr).subscribe(async cid => {
+      if (!cid) resolve() // entry probably removed in a future block
```

Comment:
This isn't behaving as expected when handling events for entries that have already been removed. IPFS reolution errors are throwing from [line 59](https://github.com/AutarkLabs/open-enterprise/pull/1189/files?file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.sol&file-filters%5B%5D=dotfile&hide-deleted-files=true#diff-446ecc1afaa88101fd8bca756c71b84aR59)

Reply:
Solved by moving the `ipfsGet` part inside an `else` block, so if `cid` is not provided, then the else condition will be skipped, this way we don't have the error anymore